### PR TITLE
fix: qodana updates

### PR DESCRIPTION
**Pull request by pgautier404**
This commit introduces some code fixes pointed out by the qodana tool. It also reconfigures qodana to suppress some false positives with as much granularity as the tool allows (which is not a lot).

### CODE REVIEW

File: src/Auth/AuthUtils.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ public static function parseAuthToken(string $authToken): object
         }
 
         try {
-            list($header, $payload, $signature) = $exploded;
+            $payload = $exploded[1];
```
Comment (poppoerika) [inline src/Auth/AuthUtils.php:26]: What does this do?

File: src/Cache/Internal/ScsDataClient.php [hunk 2]
```diff
@@ -206,13 +206,13 @@ public function set(string $cacheName, string $key, string $value, int $ttlSecon
...
             );
-            $response = $this->processCall($call);
+            $this->processCall($call);
         } catch (SdkError $e) {
             return new SetError($e);
         } catch (Exception $e) {
             return new SetError(new UnknownError($e->getMessage()));
         }
-        return new SetSuccess($response, $key, $value);
+        return new SetSuccess();
```
Comment (poppoerika) [inline src/Cache/Internal/ScsDataClient.php:215]: Sorry if I missed in your previous PRs, but we don't take any params to create an instance of the `SetSuccess` anymore?

File: src/Auth/AuthUtils.php (hunk 1 above)
Comment (pgautier404) [reply src/Auth/AuthUtils.php:26]: I only use the `$payload`, so this gets rid of a warning about the other two variables being unused.

File: src/Cache/Internal/ScsDataClient.php (hunk 2 above)
Comment (pgautier404) [reply src/Cache/Internal/ScsDataClient.php:215]: No, we don't.

**Review by poppoerika — APPROVED** [review]

**Pull request merged**